invoices/views.py

```diff
@@ -0,0 +1,55 @@
+from django.contrib.auth.mixins import LoginRequiredMixin
+from django.shortcuts import get_object_or_404
+from rest_framework.viewsets import ViewSet
+from rest_framework.response import Response
+from invoices.mixins import PaginationMixin
+from invoices.models import Invoice
+from invoices.serializers import InvoiceSerializer
+
+
+class InvoiceViewset(LoginRequiredMixin, ViewSet):
+    """ Endpoint that manages invoice detail data
+    """
+    serializer_class = InvoiceSerializer
+
+    def detail(self, *args, **kwargs):
+        serializer = self.serializer_class(get_object_or_404(
+          self.serializer_class.Meta.model, id=kwargs.get('inv_id')))
+
+        return Response(serializer.data, status=200)
+
+    def update(self, *args, **kwargs):
+        invoice = get_object_or_404(Invoice, id=kwargs.get('inv_id'))
+
+        serializer = InvoiceSerializer(data=self.request.data, instance=invoice)
+        serializer.is_valid(raise_exception=True)
+        serializer.save()
+
+        return Response(serializer.data, status=200)
+
+
+class InvoicesViewset(LoginRequiredMixin, PaginationMixin ,ViewSet):
+    """ Endpoint that manages invoice data
+    """
+    serializer_class = InvoiceSerializer
+
+    def list(self, *args, **kwargs):
+        queryset = self.serializer_class.Meta.model.objects.all()
+
+        if queryset.exists():
```

Comment:
we don't need to check if there are items on the list or now. If it is blank then we return empty list.

This becomes incorrect if you return `Response(status=200)`. because the front end expects a list